src/main/java/com/puppycrawl/tools/checkstyle/site/DescriptionMacro.java

```diff
@@ -0,0 +1,177 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2025 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.maven.doxia.macro.AbstractMacro;
+import org.apache.maven.doxia.macro.Macro;
+import org.apache.maven.doxia.macro.MacroExecutionException;
+import org.apache.maven.doxia.macro.MacroRequest;
+import org.apache.maven.doxia.sink.Sink;
+import org.codehaus.plexus.component.annotations.Component;
+
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.meta.JavadocMetadataScraper;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * A macro that inserts a description of module from its Javadoc.
+ */
+@Component(role = Macro.class, hint = "description")
+public class DescriptionMacro extends AbstractMacro {
+
+    /** New line escape character. */
+    private static final String NEWLINE = "\n";
+    /** A newline with 8 spaces of indentation. */
+    private static final String INDENT_LEVEL_8 = SiteUtil.getNewlineAndIndentSpaces(8);
+    /** A newline with 10 spaces of indentation. */
+    private static final String INDENT_LEVEL_10 = SiteUtil.getNewlineAndIndentSpaces(10);
+
+    /** The javadoc of the current module being processed. */
+    private DetailNode moduleJavadoc;
```

Comment:
Please update code to avoid both non final class fields.
The can easily be local variables.